test/expectedRate.js

```diff
@@ -522,6 +522,37 @@ contract('ExpectedRate', function(accounts) {
         assert.equal(rates[1].valueOf(), 0, "unexpected rate");
     });
 
+    it("should verify when src qty 0, network.findBestRate and network.findBestRateOnlyPermission don't revert", async function() {
+        let tokenSrcInd = 2;
+        let tokenDestInd = 1;
+        let qty = 0;
+
+        //create malicious reserve
+        let badReserve = await MaliciousReserve.new(network.address, pricing1.address, admin);
```

Comment:
don't get why reserve is called malicious?
does it attack network?
or only revert when queried with 0 quantity?

Reply:
It reverts when queried with 0 quantity